public/render_app.tsx

```diff
@@ -39,20 +39,20 @@ export const renderApp = (
     params.element
   );
 
-  const EXPECTED_BASE_PATH = `/app/${PLUGIN_ID}#`;
+  const handleLinkClick = (event: MouseEvent) => {
+    const anchorTag = (event.target as HTMLElement).closest(
+      'a'
+    ) as HTMLAnchorElement | null;
 
-  const unlistenParentHistory = params.history.listen(() => {
-    if (
-      hideInAppSideNavBar &&
-      window.location.pathname.endsWith(`/app/${PLUGIN_ID}`)
-    ) {
-      window.location.href = EXPECTED_BASE_PATH;
+    if (anchorTag && anchorTag.href.endsWith(`/app/${PLUGIN_ID}`)) {
+      window.location.href = anchorTag.href + '#';
       window.dispatchEvent(new HashChangeEvent('hashchange'));
     }
-  });
+  };
+  document.addEventListener('click', handleLinkClick);
```

Comment:
If I understand correctly, this is performing this logic for every mouse click? This could add a performance hit, and feels like a workaround to me.

Can you take a look at AD plugins, or reach out to core OSD team to get better insight on how to fix the problem?

Reply:
This workaround is done to handle below scenario:
- In the NewHomePage - workflow details page, clicking search studio from sidenav will not redirect to flow framework dashboards home page (workflows list page).
- Reason : expected url by flow framework plugin is '/app/search-studio#' but getting '/app/search-studio'

Reply:
Understandable. But I still have the same concerns as mentioned. Can you follow up with AD plugin and/or core OSD team?